content/docs/guides/continuous-delivery/spinnaker.md

```diff
@@ -0,0 +1,199 @@
+
+> Tip: If your Spinnaker Kubernetes cluster was created using Pulumi, then instead of writing this YAML, you could deploy the [Secret](https://www.pulumi.com/docs/reference/pkg/kubernetes/core/v1/secret/) resource using Pulumi too.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  # Remember the name of this secret.
+  name: pulumi-secrets
+  # The namespace of this Secret and the namespace used by the plugin should match.
+  # Learn more about Plugin Configuration in the next section.
+  namespace: spinnaker
+data:
+  # Any key/value pairs defined here will be injected as environment variables
+  # available to the container in which your Pulumi app will be run.
+  PULUMI_ACCESS_TOKEN: <base64EncodedValue>
+  AWS_ACCESS_KEY_ID: <base64EncodedValue>
+  AWS_SECRET_ACCESS_KEY: <base64EncodedValue>
+  ...
+  ...
+  # If your Pulumi app is stored in a private repo, you may specify the username/password
+  # here and then use the corresponding "key" names as placeholders in the URL you specify
+  # later in the Pipeline Stage.
+  GIT_USERNAME: <base64EncodedValue>
+  GIT_PAT: <base64EncodedValue>
+```
+
+To create this resource in your Spinnaker cluster, run:
+
+```
+kubectl apply -f secrets.yaml
+```
+
+As you can see above you can set any environment variables required for the specific providers you plan to use in your Pulumi app. Learn more about provider setup [here](https://www.pulumi.com/docs/intro/cloud-providers/).
+For private Git repos, you can save the Personal Access Token used to clone your repo into the batch job container. Then when you configure the Stage in your Spinnaker Pipeline you can reference the same key names in the URL you specify.
+For example, `https://$(GIT_USERNAME):$(GIT_PAT)@github.com/owner/repo`. The env vars will automatically resolve to the values as per your Secret resource inside the pre-configured job container.
+
+## Installing The Spinnaker Plugin
+
+> While using the Pulumi plugin is optional, doing so will help reduce the maintenance of the manual steps required to run Pulumi in Spinnaker.
+
+Installing the plugin is comprised of two steps. First, you'll need to add the relevant plugin repository to your Halyard configration using the `hal` CLI. Then, you can add the plugin itself.
+
+```
+# Add the Pulumi plugins repository for Spinnaker
+hal plugins repository add pulumi --url https://raw.githubusercontent.com/pulumi/spinnaker-plugins-repository/master/repositories.json
+
+# Add the plugin itself
+hal plugins add Pulumi.PreConfiguredJobPlugin --enabled true --version 0.0.1 --extensions pulumi.PreConfiguredJobStage
+
+# Apply the updates
+hal deploy apply
+```
+
+> Depending on your Spinnaker installation you might have to run the above commands using `sudo`.
+
+**NoteE** Before you can access the newly-added plugin in the Deck UI, you must wait for the old `orca` Pod to be replaced with a new one in your cluster. This can take anywhere from a minute to a few depending on the size of your cluster.
```

Comment:
```suggestion
**Note** Before you can access the newly-added plugin in the Deck UI, you must wait for the old `orca` Pod to be replaced with a new one in your cluster. This can take anywhere from a minute to a few depending on the size of your cluster.
```